src/sql.rs

```diff
@@ -1479,6 +1479,24 @@ CREATE INDEX devmsglabels_index1 ON devmsglabels (label);
             }
             if exists_before_update {
                 disable_server_delete = true;
+
+                // Don't disable server delete if it was on by default (Nauta):
+                if let Some(provider) = get_provider_by_id(
```

Comment:
At this point I think it makes sense to introduce `Context.get_configured_provider` in `config.rs` and replace all duplicate code with it.